Fix symlink spec on ruby 2.6

deivid-rodriguez opened the pull request:
### What was the end-user problem that led to this PR?

The problem was that we were skipping this test on ruby 2.6. Skipping tests on old versions is not much of a problem, but doing it on the latest version is opening the door for regressions. So I want to fix it.

### What was your diagnosis of the problem?

My diagnosis was that since bundler is now a default gem, that bundler version is being activated by the spec, but since we are on a 2.0.1 locked bundle, the spec runs into a version mismatch error.

### What is your fix for the problem, implemented in this PR?

My fix is to enable the `Kernel.require` rubygems monkeypatch for this case (the `bundle` gem itself), that requires default gem files without activating the specific default version, and thus not forcing end users to use that. Since... that's exactly the problem here.

### Why did you choose this fix out of the possible options?

I chose this fix because it makes sense to me.

deivid-rodriguez commented:
Nah, it's more complicated than that since the lockfile parser doesn't care about whether bundler is activated or not.

olleolleolle commented on spec/spec_helper.rb:
```diff
@@ -12,6 +12,13 @@ module Gem
   end
 end
 
+if ENV.select {|k, _v| k =~ /TRAVIS/ }.any?
```
Is this the same thing?

```suggestion
if ENV.keys.any? { |k| k.start_with?('TRAVIS_') }
```

deivid-rodriguez replied:
I think so, or we could even check for an exact value: https://docs.travis-ci.com/user/environment-variables/#default-environment-variables.

olleolleolle replied:
By the way: lots of fanmail your way for all the great work and good cheer you bring to Bundler.

deivid-rodriguez commented:
Regarding this PR, I'm not sure what to do with it, since I'm not sure whether the current failure means that the original issue can actually happen again since `bundler` became a default gem, or whether it's just not as "easy" to test it.

The current "fix" is just a workaround consisting of moving `bundler` to its own isolated location, where manipulating the `$LOAD_PATH` for it won't affect the standard library and cause issues with `$LOAD_PATH` precedence. This kind of issues (other than this failing spec) have also appeared in the past, see:

* https://bugs.ruby-lang.org/issues/15469
* https://github.com/rubygems/rubygems/issues/2188

It is true that the first of these issues was already fixed by https://github.com/bundler/bundler/pull/6941, and that might also have fixed the other issue too. But maybe it's worth bringing this to ruby-core and ask whether they think moving bundler & rubygems to an isolated location (like `RbConfig::CONFIG["sitelibdir"]`) makes sense.

@hsbt Do you have any thoughts about this?

deivid-rodriguez commented:
Yay! I managed to finally get this spec passing in a reliable way, and also made sure that it is still a regression test for the issue that originated it: https://github.com/bundler/bundler/pull/6502. :tada:

Essentially nothing changed about the issue: bundler was removing itself from the LOAD_PATH when installed in a symlinked location, and we fixed that in the PR I just linked to. The manifestation of the problem did change when bundler started being shipped as a default gem. Before, requiring `bundler/setup` would just fail under this bug, but when `bundler` is present as a default gem, it would instead "leak" to the default bundler installation rather than failing.

So, I changed the test to verify that we don't leak to a bundler installation different from the current one.

deivid-rodriguez commented on spec/runtime/setup_spec.rb:
```diff
@@ -813,32 +816,30 @@ def clean_load_path(lp)
 
-    # Can't make this pass on 2.6 since the ruby standard library has the same $LOAD_PATH
-    # entry as bundler (since it's a default gem)
-    it "should successfully require 'bundler/setup'", :ruby_repo, :ruby => "< 2.6" do
```
Ooops, unintentionally removed `:ruby_repo` from here. Will add it back.

deivid-rodriguez commented:
I can't believe this is green!! :tada:

deivid-rodriguez commented:
@bundlerbot r+
